src/main/java/rlbot/render/Renderer.java

```diff
@@ -0,0 +1,225 @@
+package rlbot.render;
+
+import com.google.flatbuffers.FlatBufferBuilder;
+import rlbot.cppinterop.RLBotDll;
+import rlbot.flat.*;
+import rlbot.flat.Color;
+
+import java.awt.*;
+import java.util.*;
+import java.util.List;
+
+public abstract class Renderer {
+
+    private final int index;
+
+    private List<Integer> renderMessageOffsets = new LinkedList<>();
+    protected FlatBufferBuilder builder;
+
+    protected Renderer(int index) {
+        this.index = index;
+    }
+
+    protected RenderPacket doFinishPacket() {
```

Comment:
Any reason not to call this `finishPacket`? Seems inconsistent with `resetPacket`

Reply:
Yep, I want the subclass NamedRenderer to be able to have a finishPacket method with public visibility, but I don't want BotLoopRenderer to have any such method exposed to bot makers. My solution was to make this protected, and following from that, I needed to have it be named differently from the finishPacket that exists in NamedRenderer.

Reply:
Awesome! Thanks.